packages/bot-web-ui/src/pages/bot-builder/quick-strategy/form-wrappers/upgraded-qs-v2/mobile-form-wrapper.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useFormikContext } from 'formik';
+import { Button, Text, ThemedScrollbars } from '@deriv/components';
+import { observer } from '@deriv/stores';
+import { localize } from '@deriv/translations';
+import { useDBotStore } from 'Stores/useDBotStore';
+import { rudderStackSendQsRunStrategyEvent } from '../../../../../analytics/rudderstack-quick-strategy';
+import { TFormValues } from '../../types';
+import StrategyTabContent from '../strategy-tab-content';
+import useQsSubmitHandler from '../useQsSubmitHandler';
+import '../../quick-strategy.scss';
+import QSStepper from './qs-stepper';
+import StrategyTemplatePicker from './strategy-template-picker';
+import { STRATEGIES } from '../../config';
+import classNames from 'classnames';
+import { QsSteps } from './trade-constants';
+
+type TMobileFormWrapper = {
+    children: React.ReactNode;
+    selected_trade_type: string;
+    setSelectedTradeType: (selected_trade_type: string) => void;
+    current_step: QsSteps;
+    setCurrentStep: (current_step: QsSteps) => void;
+};
+
+const MobileFormWrapper = observer(
+    ({ children, current_step, selected_trade_type, setCurrentStep, setSelectedTradeType }: TMobileFormWrapper) => {
+        const { isValid, validateForm, values } = useFormikContext<TFormValues>();
+        const { quick_strategy } = useDBotStore();
+        const { selected_strategy } = quick_strategy;
+        const { handleSubmit } = useQsSubmitHandler();
+        const selected_startegy_label = STRATEGIES[selected_strategy as keyof typeof STRATEGIES].label;
+        const is_verified_or_completed_step =
+            current_step === QsSteps.StrategyVerified || current_step === QsSteps.StrategyCompleted;
+        const is_selected_strategy_step = current_step === QsSteps.StrategySelect;
+
+        React.useEffect(() => {
+            validateForm();
+        }, [selected_strategy, validateForm]);
+
+        const onRun = () => {
+            rudderStackSendQsRunStrategyEvent({
+                form_values: values,
+                selected_strategy,
+            });
+            handleSubmit();
+        };
+
+        const onBack = () => {
+            setCurrentStep(QsSteps.StrategySelect);
+        };
+
+        React.useEffect(() => {
+            if (isValid && current_step === QsSteps.StrategyVerified) {
+                setCurrentStep(QsSteps.StrategyCompleted);
+            }
+            if (!isValid && current_step === QsSteps.StrategyCompleted) {
+                setCurrentStep(QsSteps.StrategyVerified);
+            }
+        }, [isValid, current_step]);
```

Comment:
Same question as the above desktop form wrapper component 🙏

Reply:
same answer above